lib/pages/recycle_task.dart

```diff
@@ -63,33 +60,53 @@ class _RecycleTaskScreenState extends State<RecycleTaskScreen> {
     });
   }
 
-  void _onTapTrashIcon(Task task) {
+  void _onTapTrashIcon(BuildContext context, Task task) {
     showDialog(
       context: context,
       builder: (context) => AlertDialog(
-          surfaceTintColor: bgColor,
-          backgroundColor: bgColor,
-          title: Text(
-            'Recover or Delete Task?',
-            style: TextStyle(color: white),
-          ),
-          actions: [
-            TextButton(
-              child: Text(
-                'Recover Task',
-                style: TextStyle(color: white),
-              ),
-              onPressed: () => context.pop(),
+        key: const Key('recover_or_delete'),
+        surfaceTintColor: bgColor,
+        backgroundColor: bgColor,
+        title: Text(
+          'Recover or Delete Task?',
+          style: TextStyle(color: white),
+        ),
+        actions: [
+          TextButton(
+            key: const Key('recycle_mark_recover'),
+            child: Text(
+              'Recover',
+              style: TextStyle(color: white),
             ),
-            TextButton(
-              child: Text(
-                'Delete Permanently',
-                style: TextStyle(color: white),
-              ),
-              onPressed: () => context.pop(),
+            onPressed: () async {
+              context
+                  .read<TaskCubit>()
+                  .updateTask(task: task, newStatus: TaskStatus.todo)
+                  .then((result) {
+                context
+                    .read<TaskCubit>()
+                    .readTasksByStatus(TaskStatus.recycled);
+                context.pop();
+              });
+            },
```

Comment:
Consider refactoring this handler to a separate function.

Reply:
done